.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,34 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Run Acceptance and Unit tests
+
+on:
+  # push:
+  #   branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip   # upgrade pip to latest version
+        pip install ".[visualization]"        # install PyProject.toml dependencies (including optional dependencies)
```

Comment:
We can't include this until it is available on main.
I do like the concept though. Maybe we should also use this for the demo and include a test run to check the demo also still works.